hpat/tests/test_series.py

```diff
@@ -1685,35 +1740,26 @@ def test_impl(S):
         S = pd.Series([73, 21, 10005, 5, 1], index=[2, 3, 4, 5, 6])
         np.testing.assert_array_equal(hpat_func(S).values, test_impl(S).values)
 
-    def test_series_nsmallest1(self):
-        def test_impl(S):
-            return S.nsmallest(4)
-        hpat_func = hpat.jit(test_impl)
-
-        m = 100
-        np.random.seed(0)
-        S = pd.Series(np.random.randint(-30, 30, m))
-        np.testing.assert_array_equal(hpat_func(S).values, test_impl(S).values)
-
-    def test_series_nsmallest_default1(self):
-        def test_impl(S):
-            return S.nsmallest()
+    def test_series_nsmallest(self):
+        def test_impl():
+            series = pd.Series([1.0, np.nan, -1.0, 0.0, 5e-324])
+            return series.nsmallest(4)
         hpat_func = hpat.jit(test_impl)
 
-        m = 100
-        np.random.seed(0)
-        S = pd.Series(np.random.randint(-30, 30, m))
-        np.testing.assert_array_equal(hpat_func(S).values, test_impl(S).values)
+        np.testing.assert_array_equal(hpat_func(), test_impl())
 
-    def test_series_nsmallest_nan1(self):
-        def test_impl(S):
-            return S.nsmallest(4)
+    def test_series_nsmallest_unboxing(self):
+        def test_impl(series):
+            return series.nsmallest()
         hpat_func = hpat.jit(test_impl)
 
-        S = pd.Series([1.0, np.nan, 3.0, 2.0, np.nan, 4.0])
-        np.testing.assert_array_equal(hpat_func(S).values, test_impl(S).values)
+        for data in test_global_input_data_numeric + [[]]:
+            series = pd.Series(data * 3)
+            np.testing.assert_array_equal(hpat_func(series), test_impl(series))
```

Comment:
```suggestion
            pandas.testing.assert_series_equal(hpat_func(series), test_impl(series))
```

Reply:
@PokhodenkoSA You are right, but functionality "in old style" doesn't work with indexes correctly. As I see `np.testing.assert_array_equal` checks only data inside of series.